lib/symboldatabase.h

```diff
@@ -744,7 +744,7 @@ class CPPCHECKLIB Function {
     }
 
 public:
-    enum Type { eConstructor, eCopyConstructor, eMoveConstructor, eOperatorEqual, eDestructor, eFunction, eLambda };
+    enum Type { eConstructor, eCopyConstructor, eMoveConstructor, eOperatorEqual, eDestructor, eFunction, eLambda, eNone };
```

Comment:
I am skeptic about adding a type `eNone` here that will never be used by Function/SymbolDatabase.

I suggest some different handling in CheckClass instead. For instance an additional error function in CheckClass for the eNone case.